src/app.test.js

```diff
@@ -1,15 +1,52 @@
-// import { createApp } from "./app";
-// import userevent from '@testing-library/user-event'
-
-test("renders the todo app", () => {
-  //   // setup
-  //   const root = ...
-  //   const container = createApp(root)
-  //   // act
-  //   const input = queryByPlaceholderText('Please enter todo here')
-  //   userevent.type(input, 'new task')
-  //   const addButton = queryByText('Add')
-  //   userevent('click', addButton)
-  //   // assert
-  expect(true).tobee(true);
+import { screen, getByText } from "@testing-library/dom";
+import userEvent from "@testing-library/user-event";
+import createApp from "./app";
+// import userEvent from "@testing-library/user-event";
+
+test("renders empty todo list when starts", () => {
+  // setup
+  const root = document.body;
+  createApp(root);
+  const todoList = screen.getByTestId("todo-list");
+  // act
+  // assert
+  expect(todoList).toBeEmptyDOMElement();
+});
+
+test("shows new item in todo list when added", () => {
+  // setup
+  const input = screen.getByPlaceholderText("Please enter todo here");
+  const addButton = screen.getByTestId("add-button");
+  const inputText = "Take wife for a walk!";
+
+  // act
+  userEvent.type(input, inputText);
```

Comment:
```js
userEvent.type(input, "Take wife for a walk!");
```

Reply:
test code should be refactored, but not in the exact same way as we'd refactor production code

better to be explicit and sometimes even duplicate the values we're asserting on ☝️